scapy/arch/windows/__init__.py

```diff
@@ -177,24 +177,46 @@ def _vbs_exec_code(code, split_tag="@"):
             yield l
     os.unlink(tmpfile.name)
 
-def _vbs_get_hardware_iface_guid(devid):
+def _get_hardware_iface_guid(devid):
+    """
+    Get the hardware interface guid for device with 'devid' number
+    or None if such interface does not exist.
+    """
+    devid = int(devid) + 1
+
+    hkey = winreg.HKEY_LOCAL_MACHINE
+    node = r"SOFTWARE\Microsoft\Windows NT\CurrentVersion\NetworkCards\{}".format(devid)
+    try:
+        key = winreg.OpenKey(hkey, node)
+        guid, _ = winreg.QueryValueEx(key, "ServiceName")
+        winreg.CloseKey(key)
+    except WindowsError:
+        return None
+    guid = guid.strip()
```

Comment:
This could be rewritten as
`return guid.strip() if guid.startswith("{") and guid.endswith("}") else None`

Reply:
I don't this so.
It changes a logic.
Correct is: 
 guid = guid.strip()
and then:
return guid if guid.startswith("{") and guid.endswith("}") else None

Reply:
You are creating a useless string:

You can rewrite it as : 
`return guid.strip() if guid.startswith("{") and guid.endswith("}\n") else None`

or (better):
```
if guid.startswith("{") and guid.endswith("}\n"):
    return guid.strip()
else:
    return None
```

Reply:
It is worst because my/previous way covers cases when guid has trailing and leading spaces
in registry (e.g. during editing by the user or if software stuck value without cleaning
leading/trailing spaces etc) but still has correct format.
In this case guid is still treated as correct (not always but very often) by software.
So string is not useless because we should strip leading/trailing spaces from guid
\_before\_ checking its format/correctness.
Additionally:
Previosly guid was obtained from console stdout (through VBS) so such playing as with "}\n"
should be performed. Now it has no sense. Worse!: it is an error because in registry is is simple string and \_has no\_ any newlines.
So please leave my code "as is" because you way/code is failed :
```return guid.strip() if guid.startswith("{") and guid.endswith("}\n") else None```
always returns None.

Reply:
More restrictive way (if we want to treat guids with leading/trailing spaces as non correct)
is to remove line:
```guid = guid.strip()```
at all.